lib/internal/streams/buffer_list.js

```diff
@@ -71,19 +71,18 @@ module.exports = class BufferList {
 
   // Consumes a specified amount of bytes or characters from the buffered data.
   consume(n, hasStrings) {
-    var ret;
-    if (n < this.head.data.length) {
+    const data = this.head.data;
+    if (n < data.length) {
       // `slice` is the same for buffers and strings.
-      ret = this.head.data.slice(0, n);
-      this.head.data = this.head.data.slice(n);
-    } else if (n === this.head.data.length) {
+      const slice = data.slice(0, n);
+      this.head.data = data.slice(n);
+      return slice;
+    } else if (n === data.length) {
```

Comment:
```suggestion
    }
    
    if (n === data.length) {
```

There is a `return` statement in the previous branch so I wonder if we can remove the `else` branch.